express-notes-api/index.js

```diff
@@ -0,0 +1,102 @@
+const express = require('express');
+const fs = require('fs');
+const data = require('./data.json');
+const app = express();
+const jsonMiddleware = express.json();
+app.use(jsonMiddleware);
+
+app.get('/api/notes', (req, res, next) => {
+  const dataNotes = [];
+  for (const object in data.notes) {
+    dataNotes.push(data.notes[object]);
+  }
+  res.send(dataNotes);
+});
+
+app.get('/api/notes/:id', (req, res, next) => {
+  const dataNotes = [];
+  for (const object in data.notes) {
+    dataNotes.push(data.notes[object]);
+  }
```

Comment:
You can use bracket notation directly instead of a loop here. You just need the `req.params.id` as a key.
```js
data.notes[req.params.id] // direct access
```